test/unit/org/apache/cassandra/utils/GeneratorsTest.java

```diff
@@ -45,4 +55,42 @@ public void dnsDomainName()
     {
         qt().forAll(Generators.DNS_DOMAIN_NAME).checkAssert(InternetDomainName::from);
     }
+
+    @Test
+    public void asciiDeterministic()
```

Comment:
I thought I found a deterministic issue with text but it ended up being an issue with IntelliJ displaying texts... these tests didn't solve any problem but did tell me that could be the issue...

by default text types now display a descriptor rather than the raw text, and lets you override and show the text with a flag... one issue is that the text displayed by IntelliJ might not be the exact string which can make repos harder....

Reply:
i think i finally found the issue... when the utf-8 strings have control chars this can cause intellij to produce bad output... I added logic to convert those to escaped chars so the string output is safe.  This also makes it more clear what chars are present as `    ` could be anything